assignment-2/lean.md

```diff
@@ -1,3 +1,96 @@
 #Lean
 
-// TODO
\ No newline at end of file
+The LEAN methodology, also known as a *Toyota Production System*, it is
+engineering methodology which has its origin in the production method
+that Toyota used to produce its cars. The methodology has spread to
+other areas, including software engineering, and many companies
+replicate the process trying to achieve the same success that Toyota
+had, although so far none has been as successful as Toyota.
+
+The methodology aims to eliminate the wastes: overburden (*muri*),
+inconsistency (*mura*) and over waste (*muda*) in the process.
+Eliminating wastes is an effective way to increase profitability. To
+achieve that, they use a system known as **JIT** (*Just-In-Time*). JIT
+means to produce the **exact** component, in the **exact** moment and
+the **exact** amount.
+
+The waste is classified in three different types:
+
+- *Muri* is the process overburden, doing things in a suboptimal way.
+  It is solved by standardising the workflow optimally.
+- *Mura* is the inconsistency of the stock of products available. It
+  is solved by a JIT inventory: there is only a little or no inventory
+  of products or components and when there is a request of a product,
+  the components are requested through a pull system in which all the
+  components recursively call the components until they reach the
+  external supplier.
+- *Muda* is the over waste in its more general meaning: all those
+  added processes or goods that are not request or have any value for
+  the customer. *Muda* can be divided into two types: *muda* type I,
+  which has no value for the customer but it is necessary for the
+  manufacturing of the product; and *muda* type 2, which has no value
+  for the customer and it is not necessary either. The aim is to
+  eliminate the *muda* type 2 but there is no straightforward way of
+  doing it because of the variety of things that can be
+  considered waste.
+
+The concept of waste is a traversal concept in other software
+development methodologies. *RUP* (Rational Unified Process) focuses on
+reducing risks and scrum focuses on delivering product. Both of them do
+not pay special attention on waste.
+
+On one hand, RUP can be more wasteful that scrum. The early stages of
+RUP (business modelling, requirements and analysis & design) take a long
+period and effort and, although they produce deliverables, they are not
+the one requested by the client. This qualifies as *muda* type I, as it
+is a waste, but it is necessary. This waste does not exist on scrum. In
+scrum, the documentation is almost not existing. This means that the
+team does not produce any other deliverables but the product that
+the client requests.
```

Comment:
An important disadvantage of having these phases, is that it will take a long time before you really see the end result. This begs the question, does anybody really know what they want until they see it?